client/state/sites/reducer.js

```diff
@@ -125,24 +127,60 @@ export function items( state = {}, action ) {
 			};
 		}
 
-		case SITE_ICON_SET: {
-			const { siteId, iconUrl } = action;
+		case SITE_SETTINGS_UPDATE:
+		case SITE_SETTINGS_RECEIVE: {
```

Comment:
You mentioned in a chat that you prefer to keep the 'icon' attribute on the site endpoint to avoid extra requests. But on the same time, you're considering here `site.icon.media_id` as the preferred source of truth (which I agree with).

Maybe we should not have to care about the extra requests here (unless it's problematic, which I don't think it is), and maybe leave the DataLayer optimize this later. 

Also, if the media_id is some general information about the site, we should add it to the site endpoint directly, don't you think?